packages/ui/src/components/va-color-indicator/VaColorIndicator.vue

```diff
@@ -1,50 +1,58 @@
 <template>
   <div
     class="va-color-indicator"
-    @click="valueComputed = !valueComputed"
+    @click="handleClick"
     :class="computedClass"
     :style="computedStyle"
   >
     <div
       class="va-color-indicator__core"
-      :style="{ ...computedStyle, backgroundColor: colorComputed }"
+      :style="computedStyle"
     />
   </div>
 </template>
 
 <script lang="ts">
-import { defineComponent, computed, PropType } from 'vue'
+import { defineComponent, computed } from 'vue'
 
 import { useColors } from '../../composables/useColor'
 import { useStateful, useStatefulProps, useStatefulEmits } from '../../composables/useStateful'
 
 export default defineComponent({
   name: 'VaColorIndicator',
-  emits: useStatefulEmits,
+  emits: [...useStatefulEmits, 'on:click'],
   props: {
     ...useStatefulProps,
-    modelValue: { type: Boolean as PropType<boolean>, default: null },
-    color: { type: String as PropType<string>, default: '' },
-    square: { type: Boolean as PropType<boolean>, default: false },
+    modelValue: { type: Boolean, default: null },
+    color: { type: String, default: '' },
+    square: { type: Boolean, default: false },
   },
   setup (props, { emit }) {
     const { valueComputed } = useStateful(props, emit)
     const { getColor } = useColors()
 
     const colorComputed = computed(() => getColor(props.color))
 
-    const computedStyle = computed(() => ({ borderRadius: props.square ? '0px' : '50%' }))
+    const computedStyle = computed(() => ({
+      borderRadius: props.square ? '0px' : '50%',
+      backgroundColor: colorComputed.value,
+    }))
 
     const computedClass = computed(() => ({
       'va-color-indicator--selected': valueComputed.value,
       'va-color-indicator--hoverable': valueComputed.value !== undefined,
     }))
 
+    const handleClick = () => {
+      valueComputed.value = !valueComputed.value
+      emit('on:click')
```

Comment:
Why do we need this emit? After all, we are already throwing emit `update:modelValue`